src/datumaro/experimental/fields.py

```diff
@@ -246,3 +246,86 @@ def image_path_field(semantic: Semantic = Semantic.Default) -> Any:
         ImagePathField instance configured with the given semantic tags
     """
     return ImagePathField(semantic=semantic)
+
+
+@dataclass(frozen=True)
+class LabelField(Field):
+    """
+    Represents a unified label annotation field that supports both single and multi-label scenarios.
+
+    This field automatically detects whether the input is a single label or multiple labels
+    and handles the conversion accordingly:
+    - Single labels: stored as Int32
+    - Multi-labels: stored as List(Int32)
+    """
+
+    semantic: Semantic
+    dtype: Any
+    multi_label: bool = False  # Flag to indicate if this field should handle multi-labels
+
+    def to_polars_schema(self, name: str) -> dict[str, pl.DataType]:
+        """Generate schema based on whether this is single or multi-label."""
+        if self.multi_label:
+            return {name: pl.List(self.dtype)}
+        else:
+            return {name: self.dtype}
+
+    def to_polars(self, name: str, value: Any) -> dict[str, pl.Series]:
+        """Convert label(s) to Polars format, handling both single and multi-label cases."""
+        if value is None:
+            return {name: pl.Series(name, [None], dtype=self.dtype)}
+
+        # Handle numpy array or list (multi-label)
+        elif self.multi_label:
+            if isinstance(value, np.ndarray):
+                value_list = value.tolist()
+            elif isinstance(value, list):
+                value_list = value
+            else:
+                raise TypeError(f"Expected value to be a list or np.ndarray, got {type(value)}")
+            return {name: pl.Series(name, [value_list], dtype=pl.List(self.dtype))}
+
+        # Handle single integer value
+        elif isinstance(value, (int, np.integer)):
```

Comment:
```suggestion
        else:
```

In this case, you can assume that we are working with a single label. No need to check the input type.